boa/src/builtins/regexp/mod.rs

```diff
@@ -502,4 +502,36 @@ impl RegExp {
 
         Ok(result)
     }
+
+    /// `RegExp.prototype[ @@search ]( string )`
+    ///
+    /// The `[@@search]` method executes a search for a match between a `this` regular expression and a string.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-regexp.prototype-@@search
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@search
+    pub(crate) fn search(this: &Value, arg_str: String, ctx: &mut Context) -> Result<Value> {
+        if !this.is_object() {
+            panic!("Not an object");
+        }
```

Comment:
According to point 2. of the spec this should throw a `TypeError`.
As an example to test this: `RegExp.prototype[Symbol.search].call(0, 'test')` in Chrome throws: `TypeError: Method RegExp.prototype.@@search called on incompatible receiver 0`.
I'm not sure `.prototype` is accessible in `boa` so I don't know if we can test this.